src/H5FDros3.h

```diff
@@ -97,13 +102,17 @@
  *
  * \var char H5FD_ros3_fapl_t::secret_key[H5FD_ROS3_MAX_SECRET_KEY_LEN + 1]
  *      A string which specifies the security key.
+ *
+ * \var char H5FD_ros3_fapl_t::session_token[H5FD_ROS3_MAX_SECRET_TOK_LEN + 1]
+ *      A string which specifies the security token.
  */
 typedef struct H5FD_ros3_fapl_t {
     int32_t version;
     hbool_t authenticate;
     char    aws_region[H5FD_ROS3_MAX_REGION_LEN + 1];
     char    secret_id[H5FD_ROS3_MAX_SECRET_ID_LEN + 1];
     char    secret_key[H5FD_ROS3_MAX_SECRET_KEY_LEN + 1];
+    char    session_token[H5FD_ROS3_MAX_SECRET_TOK_LEN + 1];
```

Comment:
There's already the `H5Pset_fapl_ros3_token` function which was intended to cover the case of passing in session tokens. I'm also generally against this type of configuration structure because it gets to the point you can't instantiate one on the stack without compiler warnings.